src/main/java/mate/academy/service/impl/AuthenticationServiceImpl.java

```diff
@@ -0,0 +1,48 @@
+package mate.academy.service.impl;
+
+import java.util.Optional;
+import mate.academy.exception.AuthenticationException;
+import mate.academy.exception.RegistrationException;
+import mate.academy.lib.Inject;
+import mate.academy.lib.Service;
+import mate.academy.model.User;
+import mate.academy.service.AuthenticationService;
+import mate.academy.service.UserService;
+import mate.academy.util.HashUtil;
+
+@Service
+public class AuthenticationServiceImpl implements AuthenticationService {
+    @Inject
+    private UserService userService;
+
+    @Override
+    public User login(String email, String password) throws AuthenticationException {
+        Optional<User> userFromDbOptional = userService.findByEmail(email);
+        if (userFromDbOptional.isEmpty()) {
+            throw new AuthenticationException("Can not authenticate user");
+        }
+        User user = userFromDbOptional.get();
+        if (user.getPassword().equals(HashUtil.hashPassword(password, user.getSalt()))) {
+            return user;
+        } else {
+            throw new AuthenticationException("Can not authenticate user");
+        }
+    }
+
+    /**
+     * We should register a new user. The new user entity will contain the email and password
+     * @param email - user email. should be unique for each user
+     * @param password - user password
+     * @return new user instance
+     */
+    @Override
+    public User register(String email, String password) throws RegistrationException {
+        if (userService.findByEmail(email).isPresent()) {
+            throw new RegistrationException("Email already in use");
+        }
+        User user = new User();
+        user.setEmail(email);
+        user.setPassword(password);
```

Comment:
Ensure that the password is hashed before being stored in the `User` object. Storing plain text passwords is a security risk.